osslab/src/app/ossdocker.py

```diff
@@ -0,0 +1,147 @@
+                requests.post(containers_url)
+            except Exception as e:
+                log.error(e)
+            return True
+        return True
+
+    def health(self):
+        containers_test = self.containers_info()
+        return {
+            "status": "OK",
+            "container_count": len(containers_test)
+        }
+
+    def run(self, args):
+        container_name = args["container_name"]
+        exist = self.get_container(container_name)
+        result = {
+            "container_name": container_name
+        }
+        if exist is not None:
+            result["container_id"] = exist["Id"]
+        else:
+            image = args["image"]
+            # ports foramt: [from, to ,from2, to2]. e.g.[9080,8080,3306,3306].Similar with 'mnts'
+            ports = args["docker_ports"] if args.has_key("docker_ports") else []
+            port_bingings = dict(zip(ports[1::2], ports[::2]))
+
+            mnts = args["mnt"] if args.has_key("mnt") else []
+            mnts_f = map(
+                lambda s: s if "%s" not in s or not args.has_key("scm") else s % args["scm"]["local_repo_path"],
+                mnts[::2])
+            mnts_t = map(lambda s: {"bind": s, "ro": False}, mnts[1::2])
+            mnt_bindings = dict(zip(mnts_f, mnts_t))
+
+            # env = args["environments"] if args.has_key("environments") else {}
+            # detach = args["detach"] if args.has_key('detach') else False
+            command = args["command"] if args.has_key('command') else None
+            stdin_open = args["stdin_open"] if args.has_key("stdin_open") else False
+            tty = args["tty"] if args.has_key("tty") else False
+            dns = args["dns"] if args.has_key("dns") else None
+            entrypoint = args["entrypoint"] if args.has_key("entrypoint") else None
+            working_dir = args["working_dir"] if args.has_key("working_dir") else None
+
+            #headers = {'content-type': 'application/json'}
+            container_config = {"Image": image, "ExposedPorts": {}}
+            for key in port_bingings:
+                container_config["ExposedPorts"][str(key) + "/tcp"] = {}
+            if mnts_f:
+                for v in mnts_f:
+                    container_config["Volumes"] = {}
+                    container_config["Volumes"][v] = {}
+            container_config["Env"] = None
+            if command is not None:
+                container_config["Cmd"] = command.split(" ")
+            container_config["OpenStdin"] = stdin_open
+            container_config["Tty"] = tty
+            container_config["Dns"] = dns
+            container_config["Entrypoint"] = entrypoint
+            container_config["WorkingDir"] = working_dir
+            container_config["AttachStdin"] = args["AttachStdin"],
+            container_config["AttachStdout"] = args["AttachStdin"],
+            container_config["AttachStderr"] = args["AttachStderr"],
```

Comment:
must check whether `AttachStdin`,`AttachStdout`,`AttachStderr` exists in `args` otherwise Exception will be thrown.